streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClientTest.java

```diff
@@ -16,30 +16,143 @@
+
+
+    @Before
+    public void before() {
+        config.put(StreamsConfig.APPLICATION_ID_CONFIG, "some_app_id");
+        config.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
+    }
+
     @Test
     public void testConfigFromStreamsConfig() {
         for (final String expectedMechanism : asList("PLAIN", "SCRAM-SHA-512")) {
-            final Properties props = new Properties();
-            props.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, "some_app_id");
-            props.setProperty(SaslConfigs.SASL_MECHANISM, expectedMechanism);
-            props.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
-            final StreamsConfig streamsConfig = new StreamsConfig(props);
+            config.put(SaslConfigs.SASL_MECHANISM, expectedMechanism);
+            final StreamsConfig streamsConfig = new StreamsConfig(config);
             final AbstractConfig config = StreamsKafkaClient.Config.fromStreamsConfig(streamsConfig);
             assertEquals(expectedMechanism, config.values().get(SaslConfigs.SASL_MECHANISM));
             assertEquals(expectedMechanism, config.getString(SaslConfigs.SASL_MECHANISM));
         }
     }
 
+    @Test
+    public void shouldAddCleanupPolicyToTopicConfigWhenCreatingTopic() throws Exception {
+        final StreamsKafkaClient streamsKafkaClient = createStreamsKafkaClient();
+        verifyCorrectTopicConfigs(streamsKafkaClient, topicConfigWithNoOverrides, Collections.singletonMap("cleanup.policy", "delete"));
+    }
+
+
+    @Test
+    public void shouldAddDefaultTopicConfigFromStreamConfig() throws Exception {
+        config.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_MS_CONFIG), "100");
+        config.put(StreamsConfig.topicPrefix(TopicConfig.COMPRESSION_TYPE_CONFIG), "gzip");
+
+        final Map<String, String> expectedConfigs = new HashMap<>();
+        expectedConfigs.put(TopicConfig.SEGMENT_MS_CONFIG, "100");
+        expectedConfigs.put(TopicConfig.COMPRESSION_TYPE_CONFIG, "gzip");
+        expectedConfigs.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_DELETE);
+        final StreamsKafkaClient streamsKafkaClient = createStreamsKafkaClient();
+        verifyCorrectTopicConfigs(streamsKafkaClient, topicConfigWithNoOverrides, expectedConfigs);
+    }
+
+    @Test
+    public void shouldSetPropertiesDefinedByInternalTopicConfig() throws Exception {
+        final Map<String, String> expectedConfigs = new HashMap<>(overridenTopicConfig);
+        expectedConfigs.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
+        final StreamsKafkaClient streamsKafkaClient = createStreamsKafkaClient();
+        verifyCorrectTopicConfigs(streamsKafkaClient, topicConfigWithOverrides, expectedConfigs);
+    }
+
+    @Test
+    public void shouldOverrideDefaultTopicConfigsWithInternalTopicConfigs() throws Exception {
```

Comment:
nit: name is a bit misleading as it sounds internal topic configs has higher priority over streams configs, but it is vice versa. We could just name it `shouldOverrideDefaultTopicConfigsFromStreamConfig`.